pkg/subvolume/subvolume.go

```diff
@@ -0,0 +1,143 @@
+/*
+Copyright 2023 The Rook Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package subvolume
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"strings"
+
+	exec "github.com/rook/kubectl-rook-ceph/pkg/exec"
+	"github.com/rook/kubectl-rook-ceph/pkg/k8sutil"
+	"github.com/rook/kubectl-rook-ceph/pkg/logging"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type fsStruct struct {
+	Name string
+	data []string
+}
+
+type subvolume struct {
+	fs     string
+	svg    string
+	subvol string
+}
+
+func List(ctx context.Context, clientsets *k8sutil.Clientsets, clusterNamespace, operatorNamespace, flag string) {
+
+	pvList, err := clientsets.Kube.CoreV1().PersistentVolumes().List(context.Background(), v1.ListOptions{})
```

Comment:
This is returning all PVs of all types, not just rbd/cephfs. It could be any pv including aws, gcp, azure, or any other storage PV. I wonder if we could filter the list based on the provider type. If not, we at least need to filter them out in the `for` loop below.